dataland-frontend/src/components/general/SlideShow.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+  <div
+    ref="slider"
+    role="list"
+    :class="slidesContainerClasses"
+    @pointerdown="dragStartCondition"
+    @touchstart="dragStartCondition"
+  >
+    <slot />
+  </div>
+  <div v-if="slideCount > 1" :class="arrowsContainerClasses">
+    <button @click="move(-1)" aria-label="Previous slide" :class="leftArrowClasses" />
+    <button @click="move(1)" aria-label="Next slide" :class="rightArrowClasses" />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onUnmounted, ref, defineEmits, toRefs } from "vue";
+
+const props = withDefaults(
+  defineProps<{
+    slidesContainerClasses: string;
+    arrowsContainerClasses: string;
+    leftArrowClasses: string;
+    rightArrowClasses: string;
+    slideCount: number;
+    initialCenterSlide?: number;
+    scrollScreenWidthLimit: number;
+    slideWidth: number;
+  }>(),
+  {
+    initialCenterSlide: 0,
+    slideWidth: 440,
+  },
+);
+
+const { slideCount, initialCenterSlide, scrollScreenWidthLimit, slideWidth } = toRefs(props);
+
+const slider = ref<HTMLElement | null>(null);
+const currentSlide = ref(0);
+const emit = defineEmits(["update:currentSlide"]);
+
+let isDragging = false;
+let startPos = 0;
+let currentTranslate = 0;
+let prevTranslate = 0;
+
+const dragStartCondition = (e: PointerEvent | TouchEvent): void => {
+  if (slideCount.value <= 1) return;
+  dragStart(e);
+};
+
+const setSliderPosition = (sliderElement: HTMLElement, animate = true): void => {
+  if (animate) sliderElement.style.transition = "transform 0.3s ease-out";
+  sliderElement.style.transform = `translate3d(${currentTranslate}px, 0, 0)`;
+};
+
+const move = (direction: number): void => {
+  if (direction === 1 && currentSlide.value < slideCount.value - 1 - initialCenterSlide.value) currentSlide.value++;
+  if (direction === -1 && currentSlide.value > 0 - initialCenterSlide.value) currentSlide.value--;
+
+  emit("update:currentSlide", currentSlide.value);
+
+  currentTranslate = currentSlide.value * -slideWidth.value;
+  if (slider.value) setSliderPosition(slider.value);
+};
+
+const dragStart = (e: PointerEvent | TouchEvent): void => {
+  // Disable dragging for window width greater than <scrollScreenWidthLimit> px, for example
+  if (scrollScreenWidthLimit.value && window.innerWidth > scrollScreenWidthLimit.value) return;
+  isDragging = true;
+  startPos = "touches" in e ? e.touches[0].pageX : e.pageX;
```

Comment:
Just a question because I find this very interesting:
I have googled that "touches"  contains information about all the touch points that are currently in contact with the touch surface. This makes sense to me since the index finger of a human of course never just touches one pixel (unlike a mouse cursor).
So picking `[0]` thakes the "first" one of these touchpoints, right.
Do you know if in general web development, there is some practice to pick the "center" of all touch points?
E. g. if someone with big finger tips presses a button with her/his thumb, maybe you want to find out what the center of all the touchpoints is.

Reply:
(If my assumption is right, there is certainly some algo that can derive the center point of all touches.  We coud use smth like this as util and reuse it all over our frontend to make it user friendly)

Reply:
In a simple web page with no multi-touch interactions, calculating the centroid of all touch points probably wouldn't add much value. For most single-touch interactions like clicking buttons or scrolling, using the first touch point is sufficient and aligns with common user expectations.

Also it could make the code more complex without offering a clear benefit for basic interactions. It might be more relevant for specialised applications, like multi-touch games or drawing apps, rather than a typical web page.

Reply:
Ok thanks for your input on this, indeed an interesting topic!